### PR TITLE
Migrate netcdf-fortran to 4.5

### PR DESCRIPTION
**Pull request by xylar**
Checklist
* [x] Used a [fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by xylar** [comment]
@ocefpaf, could you take a look at this migrator?

**Comment by xylar** [comment]
Note, there is one entry in `conda_build_config.yaml` where the package is `netcdf-fortran`:
https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/master/recipe/conda_build_config.yaml#L265
and another where it is `netcdf_fortran` (with an underscore):
https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/master/recipe/conda_build_config.yaml#L506
Maybe this is standard.  My experience with conda-forge-pinning is pretty limited. Is this likely to cause problems?

**Comment by ocefpaf** [comment]
LGTM!

**Comment by ocefpaf** [comment]
> Note, there is one entry in `conda_build_config.yaml` where the package is `netcdf-fortran`:
> [/recipe/conda_build_config.yaml@`master`#L265](https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/master/recipe/conda_build_config.yaml#L265) [](/conda-forge/conda-forge-pinning-feedstock/blob/HEAD@{2019-11-24T11:41:47Z}/recipe/conda_build_config.yaml#L265)
> and another where it is `netcdf_fortran` (with an underscore):
> [/recipe/conda_build_config.yaml@`master`#L506](https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/master/recipe/conda_build_config.yaml#L506) [](/conda-forge/conda-forge-pinning-feedstock/blob/HEAD@{2019-11-24T11:41:47Z}/recipe/conda_build_config.yaml#L506)
> Maybe this is standard. My experience with conda-forge-pinning is pretty limited. Is this likely to cause problems?

Yeah. That bit me some time ago. It is correct though.

File: recipe/migrations/netcdf-fortran-4.5.x.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+migrator_ts: 1574594786
+__migrator:
+  kind:
+    version
+  migration_number:
+    1
+  build_number:
+    1
+
+netcdf-fortran:
```
Comment (CJ-Wright) [inline recipe/migrations/netcdf-fortran-4.5.x.yaml]: This needed to be an underscore, conda build doesn't like -

**Pull request merged**